RtpTools/RtpDump.cs

```diff
@@ -247,107 +247,111 @@ void ReadBinaryFileHeader()
+            }
 
-                //The format of the item does not match the reader(which would only happen if given an unknown format)
-                if (foundFormat != m_Format)
+            //The format of the item does not match the reader(which would only happen if given an unknown format)
+            if (foundFormat != m_Format)
+            {
+                //If the the format of the entry found was binary
+                if (entry.Format < FileFormat.Text)
                 {
-                    //If the the format of the entry found was binary
-                    if (entry.Format < FileFormat.Text)
-                    {
-                        //The unexpected data consists of the supposed fileheader.
-                        if (m_FileIdentifier == null)
-                        {
-                            //Assign it
-                            m_FileIdentifier = new (unexpectedData);
 
-                            //Read the following Binary File Header
-                            ReadBinaryFileHeader();
+                      //The unexpected data consists of the supposed fileheader.
+                      if (m_FileIdentifier == null)
+                      {
+                        //Assign it
+                        m_FileIdentifier = new (unexpectedData);
 
-                            //no more data is unexpected.
-                            unexpectedData = null;
 
-                            //Remove the offset which was not related to an entry.
-                            m_Offsets.RemoveAt(offsetsCount - 1);
+                        //Read the following Binary File Header
+                        ReadBinaryFileHeader();
 
-                            //Read the entry.
-                            return ReadToolEntry();
-                        }
-                        else Media.Common.TaggedExceptionExtensions.RaiseTaggedException(unexpectedData, "Encountered a Binary file header when already parsed the header. The Tag property contains the data unexpected.");
+                        //no more data is unexpected.
+                        unexpectedData = null;
+
+                        //Remove the offset which was not related to an entry.
+                        m_Offsets.RemoveAt(offsetsCount - 1);
+
+                        //Read the entry.
+                        return ReadToolEntry();
                     }
-                    else if (unexpectedData != null) Media.Common.TaggedExceptionExtensions.RaiseTaggedException(entry, "Unexpected data found while parsing a Text format. See the Tag property of the InnerException", new Common.TaggedException<Common.MemorySegment>(unexpectedData));
+                    else if (unexpectedData != null) Media.Common.TaggedExceptionExtensions.RaiseTaggedException(unexpectedData, "Encountered a Binary file header when already parsed the header. The Tag property contains the data unexpected.");
```

Comment:
I think here you passed the data rather than the entry, is it the only change?

Reply:
Lets try to keep RtpDump and RtpTool entry exactly the same for now, refactoring for removing try/catch should not change the exception messages.

Reply:
I would prefer if you got the original files from 

https://github.com/juliusfriedman/net7mma_core/blob/master/RtpTools/RtpDump.cs

https://github.com/juliusfriedman/net7mma_core/blob/master/RtpTools/RtpToolEntry.cs

And updated them again so I can take a look, it looks like some exception messages are getting swapped in this branch.

Reply:
I looked again at this and it seems I may be incorrect but I don't want to diverge your changes, please let me know if the intention is only to remove the try/catch and what you think happened here.